src/main/java/org/folio/rest/utils/MigrationUtils.java

```diff
@@ -0,0 +1,152 @@
+    log.debug("migrate:: Migrating module: moduleTo={}, migrationModule={}", attributes.getModuleTo(), migrationModule);
+    if (StringUtils.isBlank(attributes.getModuleFrom()))
+      return Future.succeededFuture();
+
+    SemVer moduleTo = moduleVersionToSemVer(migrationModule);
+    SemVer currentModuleVersion = moduleVersionToSemVer(attributes.getModuleFrom());
+    if (moduleTo.compareTo(currentModuleVersion) > 0) {
+      log.info("migrate:: moduleTo={} was greater than currentModuleVersion={} version", moduleTo, currentModuleVersion);
+      return supplier.get();
+    }
+
+    log.info("migrate:: moduleTo={} was less or equals to currentModuleVersion={}", moduleTo, currentModuleVersion);
+    return Future.succeededFuture();
+  }
+
+  public static SemVer moduleVersionToSemVer(String version) {
+    try {
+      return new SemVer(version);
+    } catch (IllegalArgumentException ex) {
+      return new ModuleId(version).getSemVer();
+    }
+  }
+
+  public static Future<Void> updateInvoiceWithFiscalYear(Map<String, String> headers, Context vertxContext) {
+    DBClient dbClient = new DBClient(vertxContext, headers);
+    RequestContext requestContext = new RequestContext(vertxContext, headers);
+    PostgresClient pgClient = dbClient.getPgClient();
+    return pgClient.withTrans(conn -> getInvoicesWithoutFiscalYearFromDb(conn)
+      .compose(invoices -> getTransactionsByInvoiceIds(invoices, requestContext))
+      .compose(invoices -> updateInvoicesWithFiscalYearId(invoices, conn))
+      .onSuccess(v -> log.info("updateInvoiceWithFiscalYear:: Successfully invoices were updated with fiscalYear"))
+      .onFailure(t -> log.info("Error to update invoices with fiscalYear", t)));
+  }
+
+  private static Future<List<Invoice>> getInvoicesWithoutFiscalYearFromDb(Conn conn) {
+    Criteria criteria = new Criteria();
+    criteria.addField("'fiscalYearId'");
+    criteria.setOperation("IS NULL");
+    Criterion criterion = new CriterionBuilder().build();
+    criterion.addCriterion(criteria);
+
+    return conn.get(INVOICE_TABLE, Invoice.class, criterion, false)
+      .map(Results::getResults)
+      .onFailure(t -> log.error("Failed to fetch invoice with query={}", criterion.toString(), t));
+  }
+
+  private static Future<List<Invoice>> getTransactionsByInvoiceIds(List<Invoice> invoices, RequestContext requestContext) {
+    if (invoices.isEmpty()) {
+      log.info("getTransactionsByInvoiceIds:: invoices is empty");
+      return Future.succeededFuture();
+    }
+    Promise<List<Invoice>> promise = Promise.promise();
+    RestClient restClient = new RestClient();
+    // Extract the invoice IDs from the list of invoices
+    List<String> invoiceIds = invoices.stream()
+      .map(Invoice::getId)
+      .toList();
+
+    // Build a query string using the 'IN' operator
+    String query = "(sourceInvoiceId==" + "(" + String.join(" or ", invoiceIds.stream().map(id -> "\"*" + id + "*\"").toList()) + "))";
```

Comment:
Get requests has limit, so this solution can not work when we have dozens of thousands of invoices, and I think that it is real scenario, we need use chunks here, here is implementation as example
https://github.com/folio-org/mod-finance/blob/master/src/main/java/org/folio/services/transactions/CommonTransactionService.java#L46